presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorMTlsTest.cpp

```diff
@@ -0,0 +1,112 @@
+ protected:
+  explicit ArrowFlightConnectorMtlsTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : ArrowFlightConnectorTestBase(std::move(config)) {}
+
+  void setFlightServerOptions(
+      flight::FlightServerOptions* serverOptions) override {
+    flight::CertKeyPair tlsCertificate{
+        .pem_cert = readFile("./data/mtls_certs/server.crt"),
+        .pem_key = readFile("./data/mtls_certs/server.key")};
+    serverOptions->tls_certificates.push_back(tlsCertificate);
+    serverOptions->verify_client = true;
+    serverOptions->root_certificates = readFile("./data/mtls_certs/ca.crt");
+  }
+
+  void executeTest(
+      bool isPositiveTest = true,
+      const std::string& expectedError = "") {
+    std::vector<int64_t> idData = {
+        1, 12, 2, std::numeric_limits<int64_t>::max()};
+
+    updateTable(
+        "sample-data",
+        makeArrowTable({"id"}, {makeNumericArray<arrow::Int64Type>(idData)}));
+
+    auto idVec = makeFlatVector<int64_t>(idData);
+
+    auto plan = ArrowFlightPlanBuilder()
+                    .flightTableScan(velox::ROW({"id"}, {velox::BIGINT()}))
+                    .planNode();
+
+    auto runQuery = [&]() {
+      AssertQueryBuilder(plan)
+          .splits(makeSplits({"sample-data"}))
+          .assertResults(makeRowVector({idVec}));
+    };
+
+    if (isPositiveTest) {
+      runQuery();
+    } else {
+      VELOX_ASSERT_THROW(runQuery(), expectedError);
+    }
+  }
+};
+
+class ArrowFlightConnectorMtlsTest : public ArrowFlightConnectorMtlsTestBase {
+ protected:
+  explicit ArrowFlightConnectorMtlsTest()
+      : ArrowFlightConnectorMtlsTestBase(
+            std::make_shared<velox::config::ConfigBase>(
+                std::unordered_map<std::string, std::string>{
+                    {ArrowFlightConfig::kDefaultServerSslEnabled, "true"},
+                    {ArrowFlightConfig::kServerVerify, "true"},
+                    {ArrowFlightConfig::kServerSslCertificate,
+                         "./data/mtls_certs/ca.crt"},
+                    {ArrowFlightConfig::kClientSslCertificate, "./data/mtls_certs/client.crt"},
+                    {ArrowFlightConfig::kClientSslKey, "./data/mtls_certs/client.key"}})) {}
+};
+
+TEST_F(ArrowFlightConnectorMtlsTest, successfulMtlsConnection) {
+  executeTest();
+}
+
+class ArrowFlightMtlsNoClientCertTest : public ArrowFlightConnectorMtlsTestBase {
+ protected:
+  ArrowFlightMtlsNoClientCertTest()
+      : ArrowFlightConnectorMtlsTestBase(
+            std::make_shared<velox::config::ConfigBase>(
+                std::unordered_map<std::string, std::string>{
+                    {ArrowFlightConfig::kDefaultServerSslEnabled, "true"},
+                    {ArrowFlightConfig::kServerVerify, "true"},
+                    {ArrowFlightConfig::kServerSslCertificate, "./data/mtls_certs/ca.crt"}})) {}
+};
+
+TEST_F(ArrowFlightMtlsNoClientCertTest, mtlsFailsWithoutClientCert) {
+  executeTest(false, "handshake failed");
```

Comment:
is there more to the message that would indicate the problem that you could verify against? Like "Missing client certificate" ?